corehq/apps/fixtures/upload/run_upload.py

```diff
@@ -233,3 +121,200 @@ def clear_fixture_quickcache(domain, data_types):
     # the cache key that contains all types.
     global_type_ids = {dt.get_id for dt in FixtureDataType.by_domain(domain) if dt.is_global}
     get_fixture_items_for_data_types.clear(domain, global_type_ids)
+
+
+def _create_data_type(domain, table_def, replace, transaction):
+    """Processes a FixtureDataType from an _FixtureTableDefinition, which gets
+    its definition from one row in the "types" sheet in the uploaded excel sheet
+
+    Returns a tuple with
+      - (unsaved) FixtureDataType instance
+      - boolean flag to indiicate if the data type was deleted
+      - a list of errors
+    """
+    errors = []
+
+    new_data_type = FixtureDataType(
+        domain=domain,
+        is_global=table_def.is_global,
+        tag=table_def.table_id,
+        fields=table_def.fields,
+        item_attributes=table_def.item_attributes
+    )
+    try:
+        existing_data_type = FixtureDataType.fixture_tag_exists(domain, new_data_type.tag)
+        if existing_data_type:
+            data_type = existing_data_type
+        # support old usage with 'UID'
+        elif table_def.uid:
+            data_type = FixtureDataType.get(table_def.uid)
+        else:
+            data_type = new_data_type
+
+        if replace and data_type != new_data_type:
+            data_type.recursive_delete(transaction)
+            data_type = new_data_type
+
+        data_type.fields = table_def.fields
+        data_type.item_attributes = table_def.item_attributes
+        data_type.is_global = table_def.is_global
+
+        if data_type.domain != domain:
+            data_type = new_data_type
+            errors.append(
+                _("'%(UID)s' is not a valid UID. But the new type is created.")
```

Comment:
Does the user need to know this? Are they supposed to take action?

Reply:
I'm not sure what sort of action would be expected. I think it would only happen if you copied a uuid from a different domain into your ecxel sheet or tried to upload a fixture sheet from a different domain. Those could be better errors, though I'm not sure how to best introduce them